app/assets/javascripts/govuk_publishing_components/lib/govspeak/youtube-link-enhancement.js

```diff
@@ -1,50 +1,101 @@
-// = require govuk_publishing_components/vendor/jquery-ui-1.10.2.custom
-// = require govuk_publishing_components/vendor/jquery.player.min
+window.GOVUK = window.GOVUK || {}
+// linting is not happy unless YT is defined
+var YT
 
-window.GOVUK = window.GOVUK || {};
-
-(function (GOVUK, $) {
+(function (GOVUK) {
   'use strict'
 
   var YoutubeLinkEnhancement = function ($element) {
     this.$element = $element
+    // store details of all the videos from the page
+    this.playList = []
+    // store all youtube player objects
+    this.players = []
   }
 
   YoutubeLinkEnhancement.prototype.init = function () {
-    var $youtubeLinks = this.$element.find('a[href*="youtube.com"], a[href*="youtu.be"]')
+    var $youtubeLinks = this.$element[0].querySelectorAll('a[href*="youtube.com"], a[href*="youtu.be"]')
     var _this = this
+    var apiScriptInserted = false
 
-    $youtubeLinks.each(function () {
-      var $link = $(this)
+    for (var i = 0; i < $youtubeLinks.length; ++i) {
+      var $link = $youtubeLinks[i]
       // if users have disabled 'campaigns' cookie in the new cookie page settings
       // we also need to disable the youtube video embed
       if (_this.hasDisabledEmbed($link) || !YoutubeLinkEnhancement.campaignCookies()) {
         return true
       }
-
-      var videoId = YoutubeLinkEnhancement.parseVideoId($link.attr('href'))
+      // insert script only once, but only if a video in the list is set to embed
+      // and campaign cookies are enabled
+      if (!apiScriptInserted) {
+        _this.insertApiScript()
+        apiScriptInserted = true
+      }
+      var videoId = YoutubeLinkEnhancement.parseVideoId($link.getAttribute('href'))
       if (videoId) {
-        _this.embedVideo($link, videoId)
+        _this.setupVideo($link, videoId)
       }
-    })
+    }
   }
 
-  YoutubeLinkEnhancement.prototype.hasDisabledEmbed = function ($link) {
-    return $link.attr('data-youtube-player') === 'off'
+  YoutubeLinkEnhancement.prototype.playerApiReady = function () {
```

Comment:
I don't think this function (and other global state ones) should be on the prototype of YoutubeLinkEnhancement as we'd only want these run once I'd expect rather than for every instance of an embed.